src/hb-serialize.hh

```diff
@@ -469,16 +487,40 @@ struct hb_serialize_context_t
 
     assert (current);
 
-    auto& link = *current->virtual_links.push ();
-    if (current->virtual_links.in_error ())
+    if (!current->add_virtual_link(objidx))
       err (HB_SERIALIZE_ERROR_OTHER);
+  }
 
-    link.width = 0;
-    link.objidx = objidx;
-    link.is_signed = 0;
-    link.whence = 0;
-    link.position = 0;
-    link.bias = 0;
+  objidx_t last_added_child_index() const {
+    if (unlikely (in_error ())) return -1;
+
+    assert (current);
+    if (current->real_links.get_size() <= 0) {
```

Comment:
Don't use `get_size()`. That returns the byte size... Use length or just let it automatically cast to bool. What's the negative check for?